docs/src/main/asciidoc/qute-reference.adoc

```diff
@@ -1710,15 +1710,30 @@ A CDI bean annotated with `@Named` can be referenced in any template through `cd
 NOTE: `@Named @Dependent` beans are shared across all expressions in a template for a single rendering operation, and destroyed after the rendering finished.
 
 All expressions with `cdi` and `inject` namespaces are validated during build.
-
 For the expression `cdi:personService.findPerson(10).name`, the implementation class of the injected bean must either declare the `findPerson` method or a matching <<template_extension_methods,template extension method>> must exist.
-
 For the expression `inject:foo.price`, the implementation class of the injected bean must either have the `price` property (e.g. a `getPrice()` method) or a matching <<template_extension_methods,template extension method>> must exist.
 
 NOTE: A `ValueResolver` is also generated for all beans annotated with `@Named` so that it's possible to access its properties without reflection.
 
 TIP: If your application serves xref:http-reference.adoc[HTTP requests] you can also inject the current `io.vertx.core.http.HttpServerRequest` via the `inject` namespace, e.g. `{inject:vertxRequest.getParam('foo')}`.
 
+Sometimes it may be necessary to access public methods and properties of a CDI bean that is not annotated with `@Named`.
+However, if you don't control the source of the bean it is not possible to add the `@Named` annotation.
+Nevertheless, it is possible to create an intermediate CDI bean annotated with `@Named`.
+This intermediate bean can inject the bean in question and make it accessible. 
+A Java record is a very convenient way to define such an intermediate CDI bean.
+
+[source,java]
+----
+@Named <1>
+public record User(UserInfo info, @LoggedIn String username) { <2>
+}
+----
+<1> The name of the bean is defaulted to `user`. The `@Singleton` scope is added automatically.
```

Comment:
I wonder if users won't wrongly conclude that the name defaults to `user` no matter what the class is called. Perhaps we should say that it defaults to the class name lower-cased (in this case `user`)?

Reply:
Good point, or at least make it obvious from the record class - `UserData` and say that it is defaulted to `userData`

Reply:
Or add a link to the CDI spec...

Reply:
Yes, but you know how it is with users and reading specs :)
I'd just go with more complex name in the example that showcases it.